src/settings.js

```diff
@@ -857,6 +857,8 @@ var FETCH = 0; // If nonzero, enables emscripten_fetch API.
 
 var ASMFS = 0; // If set to 1, uses the multithreaded filesystem that is implemented within the asm.js module, using emscripten_fetch. Implies -s FETCH=1.
 
+var NO_PORTS = 0; // Disables usage of emscripten ports. Used internally when building system libraries.
```

Comment:
Nit: instead of `NO_PORTS = 0`, what about `BUILD_PORTS = 1`, that way we avoid `if not NO_PORTS` style double-negatives.

Though by defaulting it to `0`, we can signify in the name that not building the ports is the weirder case. So, either is probably good.